src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -74,5 +128,155 @@ public VisualStudioDiagnosticAnalyzerService(VisualStudioWorkspace workspace, ID
             // unfortunately, we had to do this since ruleset editor and us are set to use this signature
             return map.ToDictionary(kv => kv.Key, kv => (IEnumerable<DiagnosticDescriptor>)kv.Value);
         }
+
+        private void OnRunCodeAnalysisForSelectedProjectStatus(object sender, EventArgs e)
+        {
+            var command = (OleMenuCommand)sender;
+
+            // We hook up the "Run Code Analysis" menu commands for CPS based managed projects.
+            // These commands are already hooked up for csproj based projects in StanCore, but those will eventually go away.
+            var visible = TryGetSelectedProjectHierarchy(out var hierarchy) &&
+                hierarchy.IsCapabilityMatch("CPS") &&
+                hierarchy.IsCapabilityMatch(".NET");
```

Comment:
💭 Do we not already have constants somewhere for these capabilities?

Reply:
They are defined in project-system repo [here](https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectCapability.cs), but I was not sure if we want to reference a new assembly just for the constants, especially given I am not sure if this can cause any eager loading issues.